Correct fix for init.lua?

Lap76 opened the pull request.

rpworksmods committed:
Update README.md

Lap76 committed:
Merge branch 'main' of https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM

Lap76 committed:
Update init.lua

Lap76 committed:
Update init.lua

Lap76 committed:
Merge branch 'main' of https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM

Lap76 committed:
Update init.lua

Lap76 committed:
Merge branch 'main' of https://github.com/Lap76/IIS-EmergencyLS-ELS-FiveM

Lap76 committed:
Update README.md

Lap76 committed:
Update init.lua

matsn0w commented on resource/server/init.lua:
```diff
@@ -11,7 +11,7 @@ local function checkForUpdate()
 
     if not versionFile then
-        print('Couldn\'t read version file!')
+        print("Couldn\'t read version file!")
```
Can you please revert all quote changes?

matsn0w commented on resource/server/init.lua:
```diff
@@ -44,18 +44,22 @@ local function checkForUpdate()
-
-    -- this guy probably has some custom ENV var set...
-    if unix and windows then error('Couldn\'t identify the OS unambiguously.') end
+    local osType = GetConvar("os_type", "")
```
Can you refer to some documentation or otherwise prove that this is actually working?

matsn0w reviewed (changes requested):
Can you squash all 9 commits into one please?

Trko0045 commented:
any updates on this ? im on linux and ELS dosent work

Lap76 committed:
Update init.lua

Fixes error with Linux builds

Lap76 commented:
> any updates on this ? im on linux and ELS dosent work

Latest version on my GitHub should have the fix

Lap76 committed:
Update lights.lua

Attempt to fix the error "attempt to index a nil value" when using certain vehicles

Lap76 committed:
Update lights.lua

Lap76 committed:
Onesync Fix

- OneSync siren/horn sync switched to vehicle netId instead of playerid.
- Guards added for missing VCF/sound data.
- Entity existence checks added to prevent "No such entity" spam.
- Removed invalid vehicles from kjEnabledVehicles automatically.
- resetMiscs bug fixed to use misc instead of extra

Lap76 committed:
misc fixes

- OneSync siren/horn sync via vehicle netId
- local fallback for own siren activation in client/main.lua
- extra debug logging for siren resolution
- safer vehicle entity checks
- cleanup for deleted vehicle handles
- resetMiscs bugfix

Lap76 committed:
V3

- Siren state is now stored on the vehicle entity as state bag key: kjelsSiren
- Clients re-apply the custom siren when a streamed vehicle comes into scope later
- Native/default GTA sirens are muted more aggressively to avoid hearing the default siren instead of the custom one
- Keeps the previous netId-based sync and local self-hear fallback
- Retains protections against nil VCF/sounds lookups and stale vehicle entities
